exporter/splunkhecexporter/client.go

```diff
@@ -866,6 +867,31 @@ func (c *client) stop(context.Context) error {
 	return nil
 }
 
-func (c *client) start(context.Context, component.Host) (err error) {
+func (c *client) start(_ context.Context, host component.Host) (err error) {
+	httpClient, err := createHTTPClient(c.config, host, c.telemetrySettings)
+	if err != nil {
+		return err
+	}
+	c.client = httpClient
+
+	if c.config.HecHealthCheckEnabled {
+		if err = c.checkHecHealth(); err != nil {
+			return fmt.Errorf("health check failed: %w", err)
+		}
+	}
 	return nil
 }
+
+func createHTTPClient(config *Config, host component.Host, telemetrySettings component.TelemetrySettings) (*http.Client, error) {
+	maxConns := int(config.MaxConnections)
+	if config.HTTPClientSettings.MaxIdleConns == nil {
+		config.HTTPClientSettings.MaxIdleConns = &maxConns
+	}
+	if config.HTTPClientSettings.MaxIdleConnsPerHost == nil {
+		config.HTTPClientSettings.MaxIdleConnsPerHost = &maxConns
```

Comment:
I think we should deprecate `config.MaxConnections` and allow using `config.MaxIdleConns` and `config.MaxIdleConnsPerHost` with `config.MaxConnections` as a fallback. Not sure why we would need to limit the user. WDYT?

Reply:
I'm good with this, just want to know how to best document, similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16837#discussion_r1051141039

Reply:
Waiting on making sure #16837 makes it before I push more on this.

Reply:
The signalfx PR is merged now. The only needed follow-up is a deprecation warning